@xen-orchestra/backups/RemoteAdapter.mjs

```diff
@@ -742,8 +744,15 @@ export class RemoteAdapter {
     }
   }
 
-  readFullVmBackup(metadata) {
-    return this._handler.createReadStream(resolve('/', dirname(metadata._filename), metadata.xva))
+  async readFullVmBackup(metadata) {
+    const xvaPath = resolve('/', dirname(metadata._filename), metadata.xva)
+    const stream = await this._handler.createReadStream(xvaPath)
+    try {
+      stream.length = await this._handler.getSize(xvaPath)
+    } catch (error) {
+      warn(`Can't compute length of xva file`, { xvaPath, error })
+    }
```

Comment:
`fs/Abstract#createReadStream` already set `stream.length` if the remote is not encrypted.